### PR TITLE
Adds helm, kustomize and kubelogin tools to azd image

### LINKED ISSUE #3394: [Issue] azd cli github action for aks cluster: kubelogin not found in path
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
Run `azd version` and copy and paste the output here:
image: mcr.microsoft.com/azure-dev-cli-apps:latest

**Describe the bug**
Running azd in a github action with the latest version.  After provisioning the aks cluster, it fails with the error message:
  ```
(✓) Done: AKS Managed Cluster: aks-azdaks01
ERROR: deployment failed: failed invoking event handlers for 'postprovision', converting kubeconfig: exec: "kubelogin": executable file not found in $PATH,converting kubeconfig: exec: "kubelogin": executable file not found in $PATH
Error: Process completed with exit code 1.
```

**To Reproduce**
Run AZD Up on the cqgh branch of 
https://github.com/Azure-Samples/ratings-nodejs-azd-akslza

**Expected behavior**
Expect for kubelogin to be available in the path on the azd image.

### PR DESCRIPTION
**Pull request by wbreza**
Resolves #3394
Adds the following tools to the `azd` container image

- [x] Helm
- [x] Kustomize
- [x] Kubelogin

### CODE REVIEW

File: cli/Dockerfile [hunk 1]
```diff
@@ -44,6 +44,14 @@ RUN export DEBIAN_FRONTEND=noninteractive \
...
      && apt-get update \
      && apt-get install -y kubectl \
+     # Helm CLI
+     && curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 \
+     && chmod 700 get_helm.sh \
+     && ./get_helm.sh \
+     # Kustomize CLI
+     && curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash \
+     # Kubelogin CLI
+     && az aks install-cli \
```
Comment (vhvb1989) [inline cli/Dockerfile:54]: I don't think we are using this Dockerfile anymore.

We are using a base image and devcontainer features.  It looks like we can use `ghcr.io/devcontainers/features/kubectl-helm-minikube:1` from https://containers.dev/features

Comment (wbreza) [reply cli/Dockerfile:54]: Some people are still using it.  I'll update the devcontainer in addition to the dockerfile.

File: templates/common/.devcontainer/devcontainer.json/aks/nodejs/devcontainer.json [hunk 2]
```diff
@@ -29,7 +31,7 @@
         }
     },
     "forwardPorts": [3000, 3100],
-    "postCreateCommand": "",
+    "postCreateCommand": "sudo az aks install-cli",
```
Comment (wbreza) [inline templates/common/.devcontainer/devcontainer.json/aks/nodejs/devcontainer.json:34]: Installs `kubelogin`

File: cli/Dockerfile (hunk 1 above)
Comment (vhvb1989) [reply cli/Dockerfile:54]: yeah.  The universal-img we have for azd includes all the tools supported by azd. It is convenient for getting started, but it gives projects more than what they need. For example, a python+nodejs project would ended up with dotnet, java, aks, etc.
The current size of the universal-azd-img is `> 1Gb`.  It takes several minutes to be created locally and during CI/CD.

By installing only what it is needed for you pipeline, you can save gh-action minutes (which are not unlimited for everyone).

I also talked to @danieljurek in the past about stop producing and pushing the universal-img to `mcr.microsoft.com/` as it is way to heavy to be continuously pushed.  I am not sure if we did stop that or not. :)

Comment (danieljurek) [reply cli/Dockerfile:54]: We still push with updated bits on release https://mcr.microsoft.com/en-us/product/azure-dev-cli-apps/tags

**Review by vhvb1989 — APPROVED** [review]
LGTM

**Pull request closed without merging**